src/webxdc.rs

```diff
@@ -505,22 +524,19 @@ impl Context {
         Ok(())
     }
 
-    /// Pops one record of queued webxdc status updates.
-    /// This function exists to make the sqlite statement testable.
-    async fn pop_smtp_status_update(
+    /// Returns one record of the queued webxdc status updates.
+    async fn smtp_status_update_get(
         &self,
-    ) -> Result<Option<(MsgId, StatusUpdateSerial, StatusUpdateSerial, String)>> {
-        let _lock = self.sql.write_lock().await;
+    ) -> Result<Option<(MsgId, i64, StatusUpdateSerial, String)>> {
         let res = self
             .sql
             .query_row_optional(
-                "DELETE FROM smtp_status_updates
-                     WHERE msg_id IN (SELECT msg_id FROM smtp_status_updates LIMIT 1)
-                     RETURNING msg_id, first_serial, last_serial, descr",
+                "SELECT msg_id, first_serial, last_serial, descr \
+                 FROM smtp_status_updates LIMIT 1",
                 (),
                 |row| {
                     let instance_id: MsgId = row.get(0)?;
-                    let first_serial: StatusUpdateSerial = row.get(1)?;
+                    let first_serial: i64 = row.get(1)?;
```

Comment:
`first_serial` can be negative now because of the logic in `chat::resend_msgs()`:
```
"INSERT INTO smtp_status_updates (msg_id, first_serial, last_serial, descr) \
VALUES(?, ?, ?, '') \
ON CONFLICT(msg_id) \
DO UPDATE SET first_serial=min(first_serial - 1, excluded.first_serial)",
```